api/resourceserver/check_webhook.go

```diff
@@ -0,0 +1,69 @@
+package resourceserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/concourse/atc"
+	"github.com/concourse/atc/db"
+	"github.com/concourse/atc/dbng"
+	"github.com/concourse/atc/resource"
+	"github.com/tedsuo/rata"
+)
+
+// CheckResourceWebHook defines a handler for process a check resource request via an access token.
+func (s *Server) CheckResourceWebHook(pipelineDB db.PipelineDB, dbPipeline dbng.Pipeline) http.Handler {
+	logger := s.logger.Session("check-resource-webhook")
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		resourceName := rata.Param(r, "resource_name")
+		accessToken := r.URL.Query().Get("access_token")
+
+		if accessToken == "" {
+			logger.Info("malformed-request", lager.Data{"error": "missing access_token"})
+			w.WriteHeader(http.StatusBadRequest)
```

Comment:
i waffled on whether to send a 400 vs a 401. It technically is a bad request because all of the required attributes aren't present but based on your equating the token to a password then 401 make sense. So I can go either way.